aggmetric: clean up ChildrenStorage interface

RaduBerinde opened the pull request:
The interface has a `Do` function which passes an arbitrary
`interface{}` object to the function. The only thing the function can
do with that object is to pass it to `GetChildMetric`.

This dance is strange and unnecessary; we rename `Do` to `ForEach` and
change it to pass `ChildMetric` objects directly.

Informs: #144504
Epic: none
Release note: None

arjunmahishi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/144555#-:-ONzq8me8Sbe3ciGA9YA:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @aa-joshi, @Abhinav1299, and @alyshanjahani-crl)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+

RaduBerinde committed:
aggmetric: clean up ChildrenStorage interface

The interface has a `Do` function which passes an arbitrary
`interface{}` object to the function. The only thing the function can
do with that object is to pass it to `GetChildMetric`.

This dance is strange and unnecessary; we rename `Do` to `ForEach` and
change it to pass `ChildMetric` objects directly.

Informs: #144504
Epic: none
Release note: None

RaduBerinde commented:
bors r+
